src/entities/BlockNode/index.ts

```diff
@@ -39,16 +40,55 @@ export class BlockNode {
    *
    * @param args - TextNode constructor arguments.
    * @param args.name - The name of the BlockNode.
-   * @param args.children - The child nodes of the BlockNode.
+   * @param args.data - The content of the BlockNode.
    * @param args.parent - The parent EditorDocument of the BlockNode.
    * @param args.tunes - The BlockTunes associated with the BlockNode.
    */
-  constructor({ name, children, parent, tunes = {} }: BlockNodeConstructorParameters) {
+  constructor({ name, data, parent, tunes = {} }: BlockNodeConstructorParameters) {
     this.#name = name;
-    this.#children = children;
+    this.#data = data;
     this.#parent = parent;
     this.#tunes = tunes;
   }
+
+  /**
+   * Returns serialized object representing the BlockNode
+   */
+  public get serialized(): BlockNodeSerialized {
+    const serializedData = Object
+      .entries(this.#data)
+      .reduce(
+        (acc, [dataKey, value]) => {
+          if (value instanceof Array) {
+            acc[dataKey] = value.map((node) => node.serialized);
```

Comment:
please add a comment here, maybe with the example of result key and value

Reply:
also, please add a test case for it

Reply:
I've added a comment. There is a test case for TextNodes. I don't need to test the return value from node.serialized because it's a part of TextNode and FormattingNode.